python/src/ops.cpp

```diff
@@ -1410,6 +1410,72 @@ void init_ops(py::module_& m) {
       Returns:
           array: An identity matrix of size n x n.
       )pbdoc");
+  m.def(
+      "tri",
+      [](int n, int m, int k, std::optional<Dtype> dtype, StreamOrDevice s) {
+        return tri(n, m, k, dtype.value_or(float32), s);
+      },
+      "n"_a,
+      "m"_a,
+      "k"_a,
+      "dtype"_a = std::nullopt,
```

Comment:
This could just default to `float32` rather than use an `optional` type. Simpler that way.

Reply:
I should have caught that for some other ops like `eye` above. If you are up for cleaning those as well that would be nice!